packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -390,41 +400,76 @@ const Autocomplete = <
       (item: ReactElement & { children?: ReactElement[] }) =>
         [item].concat(item.children || []),
     );
-
-    // the height of an Odyssey autocomplete option item that is used to calculate height of window
-    const optionHeight = 45; //px
+    const sizeMap = useMemo(() => new Map<number, number>(), []);
 
     // The number of items (rows or columns) to render outside of the visible area for performance and scrolling reasons
     const overscanRowCount = 8;
 
-    const itemSize = useCallback(() => optionHeight, []);
-
     const gridRef = useResetCache(itemData.length);
 
+    const setItemSize = useCallback<SetItemSize>(
+      (index, size) => {
+        gridRef?.current?.resetAfterIndex(0, true);
+        sizeMap.set(index, size);
+      },
+      [gridRef, sizeMap],
+    );
+
+    const getItemSize = useCallback(
+      (index: number) => sizeMap.get(index) || 45,
+      [sizeMap],
+    );
+
+    const getListBoxHeight = useCallback(
+      (data: ReactElement[]) => {
+        // 8px of padding top/bottom applied by MUI
+        const LISTBOX_PADDING = 8;
+
+        if (data.length > 8) {
```

Comment:
I want to know what this `8` is for when skimming the code.

Can we add a comment for it?

Is it possible to set this to `Infinity` if we don't care?

Reply:
If we can't use `Infinity`, make `500` much higher.